adafruit_ina219.py

```diff
@@ -46,71 +46,77 @@
 
 # Bits
 # pylint: disable=bad-whitespace
-_READ                            = const(0x01)
+# pylint: disable=too-few-public-methods
 
 # Config Register (R/W)
-_REG_CONFIG                      = const(0x00)
-_CONFIG_RESET                    = const(0x8000)  # Reset Bit
-
-_CONFIG_BVOLTAGERANGE_MASK       = const(0x2000)  # Bus Voltage Range Mask
-_CONFIG_BVOLTAGERANGE_16V        = const(0x0000)  # 0-16V Range
-_CONFIG_BVOLTAGERANGE_32V        = const(0x2000)  # 0-32V Range
-
-_CONFIG_GAIN_MASK                = const(0x1800)  # Gain Mask
-_CONFIG_GAIN_1_40MV              = const(0x0000)  # Gain 1, 40mV Range
-_CONFIG_GAIN_2_80MV              = const(0x0800)  # Gain 2, 80mV Range
-_CONFIG_GAIN_4_160MV             = const(0x1000)  # Gain 4, 160mV Range
-_CONFIG_GAIN_8_320MV             = const(0x1800)  # Gain 8, 320mV Range
-
-_CONFIG_BADCRES_MASK             = const(0x0780)  # Bus ADC Resolution Mask
-_CONFIG_BADCRES_9BIT             = const(0x0080)  # 9-bit bus res = 0..511
-_CONFIG_BADCRES_10BIT            = const(0x0100)  # 10-bit bus res = 0..1023
-_CONFIG_BADCRES_11BIT            = const(0x0200)  # 11-bit bus res = 0..2047
-_CONFIG_BADCRES_12BIT            = const(0x0400)  # 12-bit bus res = 0..4097
-
-_CONFIG_SADCRES_MASK             = const(0x0078)  # Shunt ADC Resolution and Averaging Mask
-_CONFIG_SADCRES_9BIT_1S_84US     = const(0x0000)  # 1 x 9-bit shunt sample
-_CONFIG_SADCRES_10BIT_1S_148US   = const(0x0008)  # 1 x 10-bit shunt sample
-_CONFIG_SADCRES_11BIT_1S_276US   = const(0x0010)  # 1 x 11-bit shunt sample
-_CONFIG_SADCRES_12BIT_1S_532US   = const(0x0018)  # 1 x 12-bit shunt sample
-_CONFIG_SADCRES_12BIT_2S_1060US  = const(0x0048)  # 2 x 12-bit shunt samples averaged together
-_CONFIG_SADCRES_12BIT_4S_2130US  = const(0x0050)  # 4 x 12-bit shunt samples averaged together
-_CONFIG_SADCRES_12BIT_8S_4260US  = const(0x0058)  # 8 x 12-bit shunt samples averaged together
-_CONFIG_SADCRES_12BIT_16S_8510US = const(0x0060)  # 16 x 12-bit shunt samples averaged together
-_CONFIG_SADCRES_12BIT_32S_17MS   = const(0x0068)  # 32 x 12-bit shunt samples averaged together
-_CONFIG_SADCRES_12BIT_64S_34MS   = const(0x0070)  # 64 x 12-bit shunt samples averaged together
-_CONFIG_SADCRES_12BIT_128S_69MS  = const(0x0078)  # 128 x 12-bit shunt samples averaged together
-
-_CONFIG_MODE_MASK                = const(0x0007)  # Operating Mode Mask
-_CONFIG_MODE_POWERDOWN           = const(0x0000)
-_CONFIG_MODE_SVOLT_TRIGGERED     = const(0x0001)
-_CONFIG_MODE_BVOLT_TRIGGERED     = const(0x0002)
-_CONFIG_MODE_SANDBVOLT_TRIGGERED = const(0x0003)
-_CONFIG_MODE_ADCOFF              = const(0x0004)
-_CONFIG_MODE_SVOLT_CONTINUOUS    = const(0x0005)
-_CONFIG_MODE_BVOLT_CONTINUOUS    = const(0x0006)
-_CONFIG_MODE_SANDBVOLT_CONTINUOUS = const(0x0007)
+_REG_CONFIG                 = const(0x00)
+
+class Reset:
+    """Constants for ``reset``"""
+    RESET                   = const(0x01)       # write RESET to reset
```

Comment:
as its one bit, this could just be True or False instead of a custom class?

Reply:
I think it doesn't make any sense to use `const()` inside classes — it only works with global variables.

Reply:
Hi @ladyada, indeed this one is useless and should be removed. I will do it as soon as no more review is expected.

Hi @deshipu, I'm dupplicating what I've seen in IN260 : https://github.com/adafruit/Adafruit_CircuitPython_INA260/blob/master/adafruit_ina260.py
Using classes for constant allows to use meaning full names such as `Mode.CONTINUOUS`
I believe this is the same as for example `DigitalInOut.Direction.OUTPUT`
My understanding is that a `const()`will use less memory than a variable.

Reply:
Does it makes sense to keep this one too ?
` class BusVoltageRange:
     """Constants for ``bus_voltage_range``"""
     RANGE_16V               = const(0x00)       # set bus voltage range to 16V
     RANGE_32V               = const(0x01)       # set bus voltage range to 32V (default)
`

May be renaming `bus_voltage_range` as `bus_voltage_range_32V` and use `False`/`True` ?

Reply:
@barbudor the fact that it is used somewhere else doesn't mean that it's correct. You are just spreading a cargo cult this way.

``const()`` works by replacing every occurrence of a variable with the number itself before compiling the code. It only works with globals, because otherwise it would need to know the runtime context of the variable to be able to tell if it's really this variable or some other local variable that just happens to be named the same. It makes the code slightly faster by avoiding variable lookup.

Since the global variable still needs to be created, so that it can be accessed by any other modules that import this module, it doesn't use less memory. For it to use less memory, you would need to: 1. use global variables, as already mentioned, 2. name them with an underscore as the first character, so that they can't be used by external modules. Then, and only then, it makes sense to use ``const()``. Otherwise I would recommend removing it to avoid confusion.

Reply:
More details: https://github.com/micropython/micropython/issues/573